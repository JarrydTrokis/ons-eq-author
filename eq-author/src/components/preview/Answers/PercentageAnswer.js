import PropTypes from "prop-types";
import React from "react";

import { Field, UnitInput, Label } from "./elements";

const PercentageAnswer = ({ answer }) => (
  <Field>
    <Label description={answer.description}>{answer.label}</Label>
    <UnitInput unit="%" trailing />
  </Field>
);

PercentageAnswer.propTypes = {
  answer: PropTypes.shape({
    description: PropTypes.string,
    label: PropTypes.string,
  }).isRequired,
};

export default PercentageAnswer;
