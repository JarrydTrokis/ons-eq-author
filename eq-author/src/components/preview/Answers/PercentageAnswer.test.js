import React from "react";
import { shallow } from "enzyme";

import PercentageAnswer from "./PercentageAnswer";

describe("Percentage answer preview", () => {
  it("should render", () => {
    const wrapper = shallow(
      <PercentageAnswer
        answer={{ label: "Label", description: "Description" }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
