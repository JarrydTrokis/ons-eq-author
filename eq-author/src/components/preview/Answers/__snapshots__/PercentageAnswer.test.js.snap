// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Percentage answer preview should render 1`] = `
<Field>
  <Label
    description="Description"
  >
    Label
  </Label>
  <InputWithUnit
    trailing={true}
    unit="%"
  />
</Field>
`;
